packages/peregrine/lib/store/actions/user/asyncActions.js

```diff
@@ -69,11 +69,9 @@ export const getUserDetails = () =>
             dispatch(actions.getDetails.request());
 
             try {
-                const userDetails = await request('/rest/V1/customers/me', {
-                    method: 'GET'
-                });
+                const { data } = await fetch();
```

Comment:
This prevents other actions from calling each other as only components will have fetchers.

Reply:
Also we would want to ensure we handle the error appropriately. Does `fetch` throw an error or would we have to destructure `error` from the result?

Reply:
...you're absolutely right, and it would be nasty to have to pass query invokers for nested action calls, but it would more clearly define those action APIs.

`fetch` does `throw` on error, so this code should function as-is (though may need to destructure `{ error }` in the `catch`.